api/client/client.go

```diff
@@ -396,9 +396,17 @@ func (c *Client) setClosed() bool {
 	return atomic.CompareAndSwapInt32(&c.closedFlag, 0, 1)
 }
 
+// WithCallOptions returns a copy of the client with the given call options set.
+// This function should be used for chaining - client.WithCallOptions().Ping()
+func (c *Client) WithCallOptions(opts ...grpc.CallOption) *Client {
+	clt := *c
```

Comment:
this might break the `c.closedFlag` usage
when you call `c.Close()` and flip that flag, `clt` here won't see it
you'd need to make that field a pointer, or replace it with some other synchronization mechanism (like a channel)